src/com/google/cose/Ec2SigningKey.java

```diff
@@ -220,4 +232,92 @@ public static Ec2SigningKey parse(byte[] keyBytes) throws CborException, CoseExc
   public static Ec2SigningKey decode(DataItem cborKey) throws CborException, CoseException {
     return new Ec2SigningKey(cborKey);
   }
+
+  public byte[] sign(Algorithm algorithm, byte[] message) throws CoseException {
+    if (keyPair.getPrivate() == null) {
+      throw new CoseException("Missing key material for signing.");
+    }
+    if (algorithm != Algorithm.SIGNING_ALGORITHM_ECDSA_SHA_256) {
+      // TODO: Add support for other algorithms.
+      throw new CoseException("Unsupported algorithm.");
+    }
+
+    // Check which security provider is being used and reply appropriately.
+    // jce sign can be used to provide sign/verify functionality for bouncy castle and standard JCE
+    return new byte[0];
+  }
+
+  private byte[] jceSign(Algorithm algorithm, byte[] message, String provider) throws CoseException {
```

Comment:
I would recommend checking for a null provider and then just using the platform preference default if it's unspecified.
if (provider == null) {
...
... Signature.getInstance(algorithm.getJavaAlgorithmId());
}

Reply:
done